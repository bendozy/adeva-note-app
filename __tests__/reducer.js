import reducer, { initialState } from '../src/reducers';

describe('todos reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CREATE_NOTE', () => {
    expect(reducer(
      initialState,
      {
        type: 'CREATE_NOTE',
      },
    )).toMatchSnapshot();
  });

  it('should handle UPDATE_NOTE', () => {
    const state = {
      notes: [{
        id: 1,
        title: 'First Example',
        details: 'First Testing detail',
      }, {
        id: 2,
        title: 'Example',
        details: 'Testing detail',
      }],
      nextId: 3,
    };

    expect(reducer(
      state,
      {
        type: 'UPDATE_NOTE',
        note: {
          id: 2,
          title: 'Update Example',
          details: 'Testing detail updated',
        },
      },
    )).toMatchSnapshot();
  });

  it('should handle DELETE_NOTE', () => {
    const state = {
      notes: [{
        id: 3,
        title: 'Delete Example',
        details: 'Delete Testing detail',
      }],
      nextId: 3,
    };

    expect(reducer(
      state,
      {
        type: 'DELETE_NOTE',
        id: 3,
      },
    )).toMatchSnapshot();
  });
});
