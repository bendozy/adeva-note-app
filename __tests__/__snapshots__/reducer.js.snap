// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos reducer should handle CREATE_NOTE 1`] = `
Object {
  "nextId": 2,
  "notes": Array [
    Object {
      "id": 1,
    },
  ],
}
`;

exports[`todos reducer should handle DELETE_NOTE 1`] = `
Object {
  "nextId": 3,
  "notes": Array [],
}
`;

exports[`todos reducer should handle UPDATE_NOTE 1`] = `
Object {
  "nextId": 3,
  "notes": Array [
    Object {
      "details": "First Testing detail",
      "id": 1,
      "title": "First Example",
    },
    Object {
      "details": "Testing detail updated",
      "id": 2,
      "title": "Update Example",
    },
  ],
}
`;

exports[`todos reducer should return the initial state 1`] = `
Object {
  "nextId": 1,
  "notes": Array [],
}
`;
